### PR TITLE
Upgrade to Julia 1.3+, use JLL packages to provide binary dependencies

### PR DESCRIPTION
**Pull request by staticfloat**
Here's a POC of eliminating BinaryProvider from your package and using the new JLL packages introduced in Julia 1.3

### CODE REVIEW

**Comment by fhs** [comment]
AppVeyor Environment: julia_version=1.3; Platform: x86 is failing. I restarted the build and it's still failing.

**Comment by staticfloat** [comment]
We can wait until RC5 is released; it has a bug fix that should address that issue

**Comment by giordano** [comment]
Bump :slightly_smiling_face:  Julia 1.3 is out, the problem with Windows i686 should have been fixed.  Another CI run?

File: Project.toml [hunk 1]
```diff
@@ -3,13 +3,12 @@ uuid = "a5390f91-8eb1-5f08-bee0-b1d1ffed6cea"
...
 [deps]
-BinaryProvider = "b99e7846-7c00-51b0-8f62-c81ae34c0232"
 Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
+Zlib_jll = "83775a58-1f1d-513f-b197-d71354ab007a"
 
 [compat]
-BinaryProvider = "≥ 0.3.0"
-julia = "≥ 0.7.0"
+julia = "≥ 1.3.0"
```
Comment (giordano) [inline Project.toml]:
```suggestion
julia = "1.3.0"
```

**Comment by fhs** [comment]
Build seems to pass now. Is this ready to be merged?

**Comment by giordano** [comment]
Need to fix the `compat` section of the `Project.toml`

**Comment by giordano** [comment]
Cool, thanks!

**Pull request merged**